### DISCUSSION TITLE
a question about the implementation of OCRNet in MMSegmentation

### DISCUSSION THREAD

**Discussion by finestprogrammer** [open-mmlab/mmsegmentation#2285] [General]
Hi contributors, thank you for your awesome project. I have a question about the implementation of OCRNet. As I read in the original paper, the OCRNet pipeline contains a module called Soft Object Regions and a module called Pixel Representations. Also, the OCRNet in MMSegmentation is implemented as a cascaded encoder-decoder with a FPN head and a OCR head followed by OCRNet. Are Soft Object Regions and Pixel Representations produced by applying FPN on outputs of HRNet? Please point out if I misunderstood anything?

**Comment by xiexinch** [comment]
Hi @createhappily,
The `pixel representation` is the output of the backbone and the `soft object regions` is produced by the FCN head.

**Reply by finestprogrammer** [reply]
Thank you.
